### PR TITLE
Fix usage with jwk.

### PR DESCRIPTION
**Pull request by Manouchehri**

### CODE REVIEW

**Comment by Manouchehri** [comment]
Tested locally and it resolves #18.

```sh
git clone https://github.com/Manouchehri/cloudflare-worker-jwt.git
cd cloudflare-worker-jwt/
yarn install
yarn run build
yarn link
cd ../ # or go to whenever you're trying to test my fork from
yarn link "@tsndr/cloudflare-worker-jwt"
```

File: src/index.ts [hunk 1]
```diff
@@ -202,7 +199,11 @@ export async function sign(payload: JwtPayload, secret: string, options: JwtSign
     const partialToken = `${base64UrlStringify(_utf8ToUint8Array(JSON.stringify({ ...options.header, alg: options.algorithm })))}.${base64UrlStringify(_utf8ToUint8Array(payloadAsJSON))}`
     let keyFormat = 'raw'
     let keyData
-    if (secret.startsWith('-----BEGIN')) {
+    if (typeof secret === 'object') {
```
Comment (tsndr) [inline src/index.ts:202]: According to the type definitions `secret` can only be a `string`. This should be typed correctly.

File: src/index.ts [hunk 2]
```diff
@@ -185,9 +185,6 @@ export async function sign(payload: JwtPayload, secret: string, options: JwtSign
     if (payload === null || typeof payload !== 'object')
         throw new Error('payload must be an object')
 
-    if (typeof secret !== 'string')
-        throw new Error('secret must be a string')
-
```
Comment (tsndr) [inline src/index.ts:190]: This should be modified and not removed.

File: src/index.ts [hunk 3]
```diff
@@ -231,9 +232,6 @@ export async function verify(token: string, secret: string, options: JwtVerifyOp
     if (typeof token !== 'string')
         throw new Error('token must be a string')
 
-    if (typeof secret !== 'string')
-        throw new Error('secret must be a string')
-
```
Comment (tsndr) [inline src/index.ts:236]: This should be modified and not removed.

File: src/index.ts [hunk 4]
```diff
@@ -271,7 +269,11 @@ export async function verify(token: string, secret: string, options: JwtVerifyOp
     }
     let keyFormat = 'raw'
     let keyData
-    if (secret.startsWith('-----BEGIN')) {
+    if (typeof secret === 'object') {
```
Comment (tsndr) [inline src/index.ts:272]: According to the type definitions `secret` can only be a `string`. This should be typed correctly.

**Review by tsndr — CHANGES_REQUESTED** [review]

**Comment by tsndr** [comment]
Thank you for your contribution, just a few changes, and then we should be good to merge 😉

**Comment by neves** [comment]
Great, I just need that to validate Google JWT https://developers.google.com/identity/gsi/web/guides/verify-google-id-token
What's the estimation to merge that?

**Comment by tsndr** [comment]
Will review either today or tomorrow :)

**Comment by Manouchehri** [comment]
I ended up switching to jose, since it worked better for my specific use-case.

**Comment by neves** [comment]
@tsndr can we reopen this, I still need this feature, otherwise I can work on my fork.

**Pull request closed without merging**